app/src/main/java/com/example/vmedvediev/redditapp/comments/CommentsActivity.kt

```diff
@@ -191,25 +189,17 @@ class CommentsActivity : AppCompatActivity() {
             headerMap["User-Agent"] = username
             headerMap["X-Modhash"] = modhash
             headerMap["cookie"] = "reddit_session=$cookie"
-        
-            val call = initRetrofit(GsonConverterFactory.create()).submitComment(headerMap, "comment", postId, comment)
-            call.enqueue(object : Callback<CommentChecker> {
-                override fun onResponse(call: Call<CommentChecker>?, response: Response<CommentChecker>?) {
-                    Log.e(TAG, "onResponse: SERVER RESPONSE: ${response.toString()}")
-                    val success = response?.body()?.success
-
-                    if (success == "true") {
-                        handleSuccessPostComment()
-                    } else {
-                        handleUnsuccessPostComment()
-                    }
-                }
 
-                override fun onFailure(call: Call<CommentChecker>?, t: Throwable?) {
-                    Log.e(CommentsActivity.TAG, "onFailure: Unable to post comment: ${t?.message}")
-                    Toast.makeText(this@CommentsActivity, "An Error Occured!", Toast.LENGTH_SHORT).show()
-                }
-            })
+        launch(UI) {
+            try {
+                val success = initRetrofit(GsonConverterFactory.create())
```

Comment:
Try not to use boolean as noun. It should be like a question, e.g, `isResponseSuccessful` or `isSuccess`
Because boolean has two meanings - true or false, and you actually don't know it in runtime